jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_annotations.cpp

```diff
@@ -0,0 +1,90 @@
+    {
+      const symbol_tablet &new_symbol_table = load_java_class(
+        "ClassWithClassTypeAnnotation",
+        "./java_bytecode/java_bytecode_parser");
+
+      THEN("The annotation should store the correct type")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::ClassWithClassTypeAnnotation");
+        const std::vector<java_annotationt> &java_annotations =
+          to_annotated_type(class_symbol.type).get_annotations();
+        java_bytecode_parse_treet::annotationst annotations;
+        convert_java_annotations(java_annotations, annotations);
+        REQUIRE(annotations.size() == 1);
+        const auto &annotation = annotations.front();
+        const auto &element_value_pair = annotation.element_value_pairs.front();
+        const auto &id = to_symbol_expr(element_value_pair.value)
+          .get_identifier();
+        REQUIRE(id2string(id) == "Ljava/lang/String;");
+      }
+    }
+    WHEN("Parsing an annotation with Class value specified to primitive")
+    {
+      const symbol_tablet &new_symbol_table = load_java_class(
+        "ClassWithPrimitiveTypeAnnotation",
+        "./java_bytecode/java_bytecode_parser");
+
+      THEN("The annotation should store the correct type")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::ClassWithPrimitiveTypeAnnotation");
+        const std::vector<java_annotationt> &java_annotations =
+          to_annotated_type(class_symbol.type).get_annotations();
+        java_bytecode_parse_treet::annotationst annotations;
+        convert_java_annotations(java_annotations, annotations);
+        REQUIRE(annotations.size() == 1);
+        const auto &annotation = annotations.front();
+        const auto &element_value_pair = annotation.element_value_pairs.front();
+        const auto &id = to_symbol_expr(element_value_pair.value)
+          .get_identifier();
+        REQUIRE(id2string(id) == "B");
+      }
+    }
+    WHEN("Parsing an annotation with Class value specified to void")
+    {
+      const symbol_tablet &new_symbol_table = load_java_class(
+        "ClassWithVoidTypeAnnotation",
+        "./java_bytecode/java_bytecode_parser");
+
+      THEN("The annotation should store the correct type")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::ClassWithVoidTypeAnnotation");
+        const std::vector<java_annotationt> &java_annotations =
+          to_annotated_type(class_symbol.type).get_annotations();
+        java_bytecode_parse_treet::annotationst annotations;
+        convert_java_annotations(java_annotations, annotations);
+        REQUIRE(annotations.size() == 1);
+        const auto &annotation = annotations.front();
+        const auto &element_value_pair = annotation.element_value_pairs.front();
+        const auto &id = to_symbol_expr(element_value_pair.value)
+          .get_identifier();
+        REQUIRE(id2string(id) == "V");
```

Comment:
as above

Reply:
Now calling `java_type_to_string` and comparing it with `void_type()`.